src/software-device.cpp

```diff
@@ -254,12 +254,26 @@ namespace librealsense
     }
 
 
-    void software_sensor::set_metadata(rs2_frame_metadata_value key, rs2_metadata_type value)
+    void software_sensor::set_metadata( rs2_frame_metadata_value key, rs2_metadata_type value )
     {
+        if( ! _metadata_parsers->count( key ) )
```

Comment:
I don't think we should test this. If setting same key again overwrite the value. This is also the current behavior.

Reply:
It's necessary. But I have a better way that will be faster, I think.

Reply:
Why is it necessary? If a user wants to override a value that was previously set he should be able to do so without clearing the whole table first

Reply:
It's necessary because now the only thing saying that the metadata exists is the presence of the parser. Why do you think it requires clearing the table first? The second time the parser would already exist and only the map is changed directly.